packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -408,22 +408,7 @@ async function ensureAvailableUsers(
 
     let foundConflict = false;
     try {
-      if (
-        eventType.recurringEvent &&
-        recurringDatesInfo?.currentRecurringIndex === 0 &&
-        recurringDatesInfo.allRecurringDates
-      ) {
-        const allBookingDates = recurringDatesInfo.allRecurringDates.map((strDate) => new Date(strDate));
-        // Go through each date for the recurring event and check if each one's availability
-        // DONE: Decreased computational complexity from O(2^n) to O(n) by refactoring this loop to stop
-        // running at the first unavailable time.
-        let i = 0;
-        while (!foundConflict && i < allBookingDates.length) {
-          foundConflict = checkForConflicts(bufferedBusyTimes, allBookingDates[i++], duration);
-        }
-      } else {
-        foundConflict = checkForConflicts(bufferedBusyTimes, input.dateFrom, duration);
-      }
+      foundConflict = checkForConflicts(bufferedBusyTimes, input.dateFrom, duration);
```

Comment:
The change looks good. I was already removing it as part of https://github.com/calcom/cal.com/pull/11774/files

But the interesting thing is that I was thinking that it was dead code but actually it was failing some bookings. 
Did some investigation as to how just enabling a calendar was failing Recurring bookings

- When Calendar is enabled, we don't just fetch the busyTimes for the current slot being booked but instead we fetch till the end of the month. This isn't true for busyTimes fetched from DB. There it's just for the range of the timeslot being booked. The behaviour [changed here 5 months back](https://github.com/calcom/cal.com/pull/9282)
- Due to this, when a single timeslot was checked for conflicts with busytimes, it was actually checking conflicts with busyTime of a recurring timeslot that was just booked in the previous iteration and thus it was always failing.


So, great find @CarinaWolli 👏